core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -663,7 +673,7 @@ class SnappyParser(session: SnappySession)
     SELECT ~ (DISTINCT ~> trueFn).? ~
     (namedExpression + commaSep) ~
     (FROM ~ ws ~ relations).? ~
-    (WHERE ~ expression).? ~
+    (WHERE ~ (test(setTokenize) ~ expression ~ test(setNoTokenize))).? ~
```

Comment:
LGTM - I think means "Looks good to merge" :-)

Initially tokenization was working on all parts of the query, I mean the parsing was all proper and all the literals were getting properly captured. Generated code was also not the problem. The problem was coming in the analyzer phase where it was not able to understand the new node type and expected very definite things against certain clauses. Limit anything other than integer against a limit clause will fail. Since we did not want to change the analyzer phase code so reduced the scope to WHERE clause of queries -- nested to any level ( should cover the majority use cases )